internal/hardware/validator.go

```diff
@@ -63,31 +68,57 @@ func isNvme(name string) bool {
 	return strings.HasPrefix(name, "nvme")
 }
 
-func ListValidDisks(inventory *models.Inventory, minSizeRequiredInBytes int64) []*models.Disk {
-	var disks []*models.Disk
-	for _, disk := range inventory.Disks {
-		if disk.SizeBytes >= minSizeRequiredInBytes && funk.ContainsString([]string{"HDD", "SSD"}, disk.DriveType) {
-			disks = append(disks, disk)
-		}
+// DiskIsEligible checks if a disk is relevant for installation by testing
+// it against a list of predicates. Also returns all the reasons the disk
+// was found to be not eligible
+func (v *validator) DiskIsEligible(disk *models.Disk) (eligible bool, notEligibleReasons []string) {
+	if minSizeBytes := gbToBytes(v.MinDiskSizeGb); disk.SizeBytes < minSizeBytes {
+		notEligibleReasons = append(notEligibleReasons,
+			fmt.Sprintf("Disk is too small (disk only has %s, but %s are required)", humanize.Bytes(uint64(disk.SizeBytes)), humanize.Bytes(uint64(minSizeBytes))))
+	}
+
+	if allowedDriveTypes := []string{"HDD", "SSD"}; !funk.ContainsString(allowedDriveTypes, disk.DriveType) {
+		notEligibleReasons = append(notEligibleReasons,
+			fmt.Sprintf("Drive type is %s, it must be one of %s.",
+				disk.DriveType, strings.Join(allowedDriveTypes, ", ")))
 	}
 
+	eligible = len(notEligibleReasons) == 0
+
+	return
+}
+
+func (v *validator) ListValidDisks(inventory *models.Inventory) []*models.Disk {
+	eligibleDisks := funk.Filter(inventory.Disks, func(disk *models.Disk) bool {
+		if !disk.InstallationEligibility.Eligible && len(disk.InstallationEligibility.NotEligibleReasons) == 0 {
```

Comment:
Please add a todo to remove it in the next version, and open a ticket for it.

Reply:
Gladly